Kconfig.zephyr

```diff
@@ -82,9 +82,15 @@ config HAS_FLASH_LOAD_OFFSET
 	  This option is selected by targets having a FLASH_LOAD_OFFSET
 	  and FLASH_LOAD_SIZE.
 
+config USE_CODE_PARTITION
+	bool "link into code-partition"
+	depends on HAS_FLASH_LOAD_OFFSET
+	help
+	  When selected application will be linked into chosen code-partition.
+
 config FLASH_LOAD_OFFSET
 	hex "Kernel load offset"
-	default $(dt_hex_val,DT_CODE_PARTITION_OFFSET) if BOOTLOADER_MCUBOOT
+	default $(dt_hex_val,DT_CODE_PARTITION_OFFSET) if USE_CODE_PARTITION
```

Comment:
Why not add a new function in `kconfigfunctions.py` that conditionally returns `DT_CODE_PARTITION_OFFSET` or 0? is it not possible?

Reply:
It's better to write Kconfig than python, isn't it?

Not sure if I understood the problem/motivation here.

Reply:
IMO it is more legible how it is implemented now.

@carles: were you thinking about: 
`default $(dt_hex_val1_or_val2, DT_CODE_PARTITION_OFFSET, 0,  USE_CODE_PARTITION)`

probably about somethings different

Reply:
nrF9160 NS board has, also, a similar setting, i.e. force using code partition.
I wonder if we could integrate this solution to the 9160 configuration, too